source/common/http/utility.cc

```diff
@@ -57,11 +57,12 @@ Utility::QueryParams Utility::parseQueryString(const std::string& url) {
     if (end == std::string::npos) {
       end = url.size();
     }
+    absl::string_view param(url.c_str() + start, end - start);
```

Comment:
Is the idea here to avoid a bug where we could see for the `=` beyond the next query param? If so, can you add a test for this?

Reply:
Yes, prior to this, `parseQueryString` had a bug whereby it would interpret  `x&y=z` as having two key-value variables:
* key=`y`, value=`z`
* key=`x&y`, value=`z`

This PR contains a test for that, on [line 1079](https://github.com/envoyproxy/envoy/pull/2258/commits/753d36950fa423561f4231a90a778c2aa236d49e#diff-fa2430fa87027867f55fc07dc4412eb3R1079) of config_impl_test.cc.